schema.hh

```diff
@@ -635,6 +636,7 @@ private:
         // The flag is not stored in the schema mutation and does not affects schema digest.
         // It is set locally on a system tables that should be extra durable
         bool _wait_for_sync = false; // true if all writes using this schema have to be synced immediately by commitlog
+        std::shared_ptr<dht::i_partitioner> _partitioner;
     };
```

Comment:
Why are you starting with std::shared_ptr? Not even seastar::shared_ptr?

Reply:
It's just an intermediate step to break the cyclic dependency. It's replaced with const& in the following patches. I tried to explain it both on commit message and PR description but apparently it didn't work.

Reply:
Ok, but it's just strange it uses std::shared_ptr<> and <memory> instead of seastar::shared_ptr?

Reply:
I'm good at complaining that changelogs are missing descriptions, but pretty bad at actually reading them.

Agree you should use a regular shared_ptr, it's just strange.